feat(instrumentation): add option to overwrite otlp service name/namespace/version with env var (#29583)
renovatebot/renovate: docs/usage/opentelemetry.md

To activate the instrumentation, you must set the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable.
This variable controls the endpoint for the telemetry data.
Once this endpoint is set, you can use all environment variables listed in the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md).

> You can also set the following environment variables:
> OTEL_SERVICE_NAME: to control the service name that will be emitted in traces, defaults to `renovate`
> OTEL_SERVICE_NAMESPACE: to control the service namespace that will be emitted in traces, defaults to `renovatebot.com`
> OTEL_SERVICE_VERSION: to control the service version that will be emitted in traces, defaults to using the pkg version

Reviewer: Let's use a list for this, and `monospace` the variable names:

```suggestion
You can also set the following environment variables:

- `OTEL_SERVICE_NAME`: to control the service name that will be emitted in traces, defaults to `renovate`
- `OTEL_SERVICE_NAMESPAC`E: to control the service namespace that will be emitted in traces, defaults to `renovatebot.com`
- `OTEL_SERVICE_VERSION`: to control the service version that will be emitted in traces, defaults to using the pkg version
```

Author: fixed!

To activate the instrumentation, you must set the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable.
This variable controls the endpoint for the telemetry data.
Once this endpoint is set, you can use all environment variables listed in the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md).
You can also set the following environment variables:

- `OTEL_SERVICE_NAME`: to control the service name that will be emitted in traces, defaults to `renovate`

> - `OTEL_SERVICE_NAMESPAC`E: to control the service namespace that will be emitted in traces, defaults to `renovatebot.com`

Reviewer 2: ```suggestion
- `OTEL_SERVICE_NAMESPACE`: to control the service namespace that will be emitted in traces, defaults to `renovatebot.com`
```

Author: fixed!

> - `OTEL_SERVICE_VERSION`: to control the service version that will be emitted in traces, defaults to using the pkg version

Reviewer: ```suggestion
- `OTEL_SERVICE_VERSION`: to control the service version that will be emitted in traces, defaults to using the `pkg` version
```

@secustor is this `pkg` thing something we should monospace? I'm not sure, so that's why I'm asking. 😄

Reviewer 2: It is basically the release version, then it is probably better call it that. 
```suggestion
- `OTEL_SERVICE_VERSION`: to control the service version that will be emitted in traces, defaults to using the release version of Renovate
```
@HonkingGoose how does this sound to you?

Reviewer: Your suggestion is _way_ easier to understand!

If this is good-enough detail for Renovate users to use the feature, we should use your suggestion.
